### ISSUE TITLE
Broken for building for Mac

### ISSUE THREAD

**Issue by PythonBlue** [FrictionalGames/AmnesiaTheDarkDescent#4]
Just so other people who want to build for Mac are aware, it won't work at this time with this given code. Granted, my computer is running Catalina (needless to say, I was hoping to make a 64-bit build), but none of the many errors I encountered were related to the architecture.

The easy part was fixing the Steam library reference mentioned in another issue here. More complex is a missing file called Launcher_Prefix.pch and a folder called "tests" when using CMake just to generate an XCode project, even if I ended up resorting to the same solution as for the Steam library.

The rest of the problems were the result of attempting to actually build, both in XCode and with makefiles. Whether it was permissions issues (which I could usually but not always overcome by running XCode as a superuser), not recognizing the standard C++ library, or, in the furthest I ever got to a successful build, lots of linker errors relating to the FBX SDK, I spent hours trying to build, with no real luck. So yeah, I'll need to wait until/unless the missing files are added and verify that Frictional was able to build using this exact code and nothing else before I'm ready to try again...

**Comment by PythonBlue** [comment]
Update: after a lot of effort, I managed to get through most of the problems except for the FBX SDK library, which, judging from the linker errors I get, seems to be prebuilt for 32-bit. Tried out compiling with a more modern FBX SDK only to get different linker errors in the end, even if fewer.

**Comment by PythonBlue** [comment]
Update #2: finally got it to work somewhat. It'll take a lot of code tweaking, so to save other Mac users some time and effort, here are my discoveries:

*Like I said, the FBX SDK needs to be updated. I finally got it to work using version 2014.2 of that library.

*The second most recurring thing I had to comment out for a successful build were a single function related to the gamepad, which means I can't guarantee that will work at this time, at least with my own attempt.

*Some of the editors will not build because it requires some code in the missing "tools" folder, which will likely affect everyone regardless of platform. Hopefully Frictional will address this.

*The build of the game itself was eventually successful, but running it initially proved misleading: for some reason, it doesn't create the Frameworks directory for the new App for me, at least. Recreate that folder, using the old 32-bit builds as reference for what should be in there, and you should be all set!

**Comment by Mudbill** [comment]
It built for you on 64-bit? That's some pretty good progress considering it doesn't exist currently.
IIRC the earlier attempts at compiling for 64-bit resulted in at least very botched physics in the game, since the Newton library had to be replaced with a 64-bit equivalent.

**Comment by PythonBlue** [comment]
Yeah, it did build at all. Course, again, the problem becomes how it plays levels. When the first thunder crack happens in 00_rainy_hall, the flicker and the thunder sound persist nonstop, and I also can’t move from that point onward. Tried loading two other levels for testing: the Zimmerman memory seemed to be fine while the Nave crashed the game on load. Granted, I was testing my mod, which used a few FBX models, but if the reason the Nave wouldn’t load is because of that, then it’s all the more reason to find a 64-bit build of the FBX 2012.2 SDK.

**Comment by PythonBlue** [comment]
Update 4: it appears the problem with the Nave, thankfully, is irrelevant to my use of FBX files: tried another modded level that makes use of one, and that plays fine. Still, it is puzzling why the Nave crashes and the Rainy Hall virtually freezes in game...I can attempt to make a pull request so that people can compare. Hopefully nothing I changed or commented out is the problem.

**Comment by PythonBlue** [comment]
Well, I think I found out why the Rainy Hall has no movement, and probably other issues. I’m hoping it’s not specific to my platform so that it can be addressed more quickly, but it appears the level-specific script functions that make use of non-void return values (such as RandFloat) do not work. I since made a pull request with my own repository, so hopefully, if it is just me, others can diagnose what’s going wrong.
